platform/mbed_retarget.cpp

```diff
@@ -261,7 +261,7 @@ MBED_WEAK FileHandle *mbed::mbed_override_console(int fd)
 
 static FileHandle *default_console()
 {
-#if DEVICE_SERIAL
+#if MBED_CONF_STDIO_SERIAL && MBED_CONF_TARGET_CONSOLE_UART && DEVICE_SERIAL
```

Comment:
Shouldn't this be MBED_CONF_PLATFORM_STDIO_SERIAL ?
Also why adding both checks here ? What would happen if only one of the configurations is set to false ? 
I think maybe ... 
```#if MBED_CONF_STDIO_SERIAL && `DEVICE_SERIAL```
would be enough in this file here

Reply:
I'm trying to make a subtle distinction between "target has a UART console interface" and "application wants to use UART console interface if available".

I think you're right, this probably isn't worth splitting into two options. If it is a target option, then target can have it true or false, and an application can either leave it unmodified if it wants to use it if available, or set it to false if not.

This would make it consistent with the existing `target.console-uart-flow-control`.